src/services/diagram-creation-service/DiagramCreationService.ts

```diff
@@ -19,6 +26,42 @@ export class DiagramCreationService implements IDiagramCreationService {
     return diagram;
   }
 
+  private async _renameDiagram(xml: string, name: string): Promise<string> {
+    const modeler: IBpmnModeler = new bundle.modeler({});
+
+    modeler.importXML(xml, (error: any) => {
+      // Do nothing for now
+    });
+
+    const promise: Promise<string> = new Promise((resolve: Function): void => {
```

Comment:
```suggestion
    const promise: Promise<string> = new Promise((resolve: Function, reject: Function): void => {
```

See below as to why.